scrapy/commands/__init__.py

```diff
@@ -178,6 +185,16 @@ def process_options(self, args: List[str], opts: argparse.Namespace) -> None:
             )
             self.settings.set("FEEDS", feeds, priority="cmdline")
 
+    def validate_feed_exporter(self, opts):
+        feed_eport_key = "scrapy.extensions.feedexport.FeedExporter"
+
+        if opts.output:
+            extensions = build_component_list(self.settings.getwithbase("EXTENSIONS"))
+            if feed_eport_key not in extensions:
```

Comment:
I think we need to also support subclasses of the built-in extension here. It's not trivial but should be possible (`load_object` + `issubclass` I think, like in `scrapy.core.engine.ExecutionEngine._get_scheduler_class()`).

Reply:
@wRAR,

`build_component_list(self.settings.getwithbase("EXTENSIONS"))` returns below mentioned extensions. The  `load_object` + `issubclass` is not working with `scrapy.extensions.telnet.TelnetConsole` as it gives `Exception: The installed reactor (twisted.internet.selectreactor.SelectReactor) does not match the requested one (twisted.internet.asyncioreactor.AsyncioSelectorReactor)` error.

```
'scrapy.extensions.corestats.CoreStats', 
'scrapy.extensions.telnet.TelnetConsole',
'scrapy.extensions.memusage.MemoryUsage', 
'scrapy.extensions.memdebug.MemoryDebugger', 
'scrapy.extensions.closespider.CloseSpider', 
'scrapy.extensions.logstats.LogStats', 
'scrapy.extensions.spiderstate.SpiderState', 
'scrapy.extensions.throttle.AutoThrottle'
```

Reply:
I don't think this should happen, how can I reproduce this?

Reply:
@wRAR,

custom_extension.py
```
import logging
from scrapy.extensions.feedexport import FeedExporter

logger = logging.getLogger(__name__)

class CustomFeedExporter(FeedExporter):
    def __init__(self, crawler):
        super().__init__(crawler)

    def item_scraped(self, item, spider):
        super().item_scraped(item, spider)
        logger.info('==> Custom FeedExporter')
```

settings.py
```
EXTENSIONS = {
    custom_extension.CustomFeedExporter": 500
}
TWISTED_REACTOR = "twisted.internet.asyncioreactor.AsyncioSelectorReactor"
```

spider
```
import scrapy

class BooklistItem(scrapy.Item):
    url = scrapy.Field() 
    title = scrapy.Field() 
    price = scrapy.Field()
             
class BookSpider(scrapy.Spider):
    name = "book"
    allowed_domains = ["books.toscrape.com"]
    start_urls = ["http://books.toscrape.com/"]

    def parse(self, response):
        for article in response.css('article.product_pod'): 
            book_item = BooklistItem( 
                url=article.css("h3 > a::attr(href)").get(), 
                title=article.css("h3 > a::attr(title)").extract_first(), 
                price=article.css(".price_color::text").extract_first(), 
            )   
            yield book_item
```

updated validate_feed_exporter definition
```
   def validate_feed_exporter(self, opts):
        if opts.output:
            extensions = build_component_list(self.settings.getwithbase("EXTENSIONS"))
            
            from scrapy.extensions.feedexport import FeedExporter
            from scrapy.utils.misc import load_object
            raise_warning = True

            for ext in extensions:           
                ext_cls = load_object(ext)
                if issubclass(load_object(ext_cls), FeedExporter):
                    raise_warning = False
                    break

            if raise_warning:
                logger.warning(
                    "Either 'FeedExporter' extension or its subclass must be enabled for Feed Exports to work."
                )
```

Reply:
@wRAR,
Were you able to reproduce this error?

Reply:
#6432